bin/bulk-update-apps.mjs

```diff
@@ -0,0 +1,114 @@
+#!/usr/bin/env node
+
+/**
+ * @file
+ * You can use this script to bump the version of many apps with the same changes.
+ * You need to specify a hash for a git commit along with a message that describes the changes made to the apps.
+ *
+ * The apps that were changed in the git commit are bumped to a new minor version,
+ * and their changelogs are updated with the message you provided.
+ *
+ * Run it like this:
+ * node bin/bulk-update-apps.mjs GITHASH CHANGELOGMESSAGE
+ *
+ * Example command:
+ * node bin/exempt-lint.mjs 29ced17i7 'Minor code improvements'
+ *
+ * You can also run it in output mode like this:
+ * node bin/bulk-update-apps.mjs GITHASH --output
+ *
+ * This mode doesn't make any changes to your files, it outputs the ID's of all apps that would be bumped.
+ */
+
+import { exec } from "node:child_process";
+import fs from "node:fs/promises";
+
+const commitHash = process.argv[2];
+if (!commitHash || commitHash === "--output") {
```

Comment:
If the user passes an extra argument, we won't catch it, e.g.
```bash
$ node bin/bulk-update-apps.mjs abc123 'message' --output --help
```
or even:
```bash
$ node bin/bulk-update-apps.mjs abc123 they accidentally forgot to quote the changelog message

# creates changes with "they" in the changelog
```

Then general way to parse arguments is either with a library or a loop, this is simple enough for a loop, e.g.
```javascript
for (let i = 2; i < process.argv.length; i++) {
    const arg = process.argv[i];

    if (arg === "--output") {
        outputFlag = true;
    } else if (!commitHash) {
        commitHash = arg;
    } else if (!changelogMessage) {
        changelogMessage = arg;
    } else {
        usage();
    }
}

if (!commitHash || !changelogMessage) usage();

function usage() {
    // print program usage to stderr and exit with an error
}
```

This also allows flexibility with where flag arguments (e.g. `--output`) can be placed (they traditionally come before positional arguments)

I think that command line programs shouldn't throw an error to exit - it leaves an ugly stack trace and exposes implementation details, they should log a message to the console with help on how the user can correct their error and exit with an exit code, but I don't mind if you want to make `usage()` throw an error

Reply:
> I think that command line programs shouldn't throw an error to exit

Maybe you're right, but throwing an error is just an easy way to make sure the program stops, and it gives us more information about what went wrong. Once again I will argue that the "users" are basically only you and me.